### PR TITLE
Use WASM commands running in webworker

### PR DESCRIPTION
**Pull request by ianthomas23**
This is an experiment using POSIX commands compiled to WebAssembly via emscripten instead of the current implementation which uses commands written in bespoke TypeScript. Here is an example:

https://github.com/jupyterlite/terminal/assets/580326/38ef0c9a-a9dc-4344-a09a-a8dd16aeeb31

It includes all of [coreutils](https://en.wikipedia.org/wiki/List_of_GNU_Core_Utilities_commands) as a single JavaScript module with embedded WASM. The build system for that is based on the marvellous [biowasm](https://github.com/biowasm/biowasm). It uses its own emscripten filesystem so is not yet integrated with the JupyterLite file browser contents (edit: it is now). There is no support for input or output redirection, pipes, or setting env vars, or many other things yet.

I have temporarily removed the `cockle` dependency and just put the JS files (`fs.js` and `coreutils.js`) in this repo for demo purposes. Eventually they will be in `cockle` (or an equivalent) and we will need to use much of the existing `cockle` functionality for command-line editing and so on. The actual JS/WASM files are built by hand at this stage but obviously that needs to be automated.

Because of global state in each command, the code recreates a new instance of the `coreutils` module for each command. This takes about 30 ms the first time, and less that 20 ms thereafter on my dev system (see log messages in console). The separate `fs.js` module is just for the emscripten filesystem which lasts the duration of the terminal and is proxied into each command just before they are run.

This reuses the WebWorker code from PR #13.

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks for looking into this!

> The actual JS/WASM files are built by hand at this stage but obviously that needs to be automated.

Out of curiosity, what is the process for building these files?

**Comment by ianthomas23** [comment]
> > The actual JS/WASM files are built by hand at this stage but obviously that needs to be automated.
> 
> Out of curiosity, what is the process for building these files?

It is the classic UNIX `configure` and `make` combination but using `emscripten`. A bit like https://github.com/emscripten-forge/recipes/blob/main/recipes/recipes_emscripten/jpeg/build.sh but with more flags and some manual ~hacks~ modifications. And it is done in an Ubuntu docker container to build against the appropriate Linux system headers and with a particular version of the `emsdk`.

**Comment by jtpio** [comment]
OK thanks.

Would it be possible to post the commands on this PR, to be able to more easily build the files locally if needed?

**Comment by ianthomas23** [comment]
> Would it be possible to post the commands on this PR, to be able to more easily build the files locally if needed?

Not yet, I need to do more on it before I am happy with it.

**Comment by ianthomas23** [comment]
This now connects to the file browser using a `DriveFS`.

https://github.com/jupyterlite/terminal/assets/580326/ba6d42f3-5874-4aff-a995-1dd8e4a07a7c

It breaks some things such as `cp` creates the new file but doesn't copy the contents, but as I haven't dealt with input/output redirection yet this probably isn't surprising.

File: package.json [hunk 1]
```diff
@@ -30,12 +30,13 @@
         "url": "https://github.com/jupyterlite/terminal.git"
     },
     "scripts": {
-        "build": "jlpm build:lib && jlpm build:labextension:dev",
+        "build": "jlpm build:lib && jlpm build:worker && jlpm build:labextension:dev",
         "build:prod": "jlpm clean && jlpm build:lib:prod && jlpm build:labextension",
```
Comment (jtpio) [inline package.json:34]: Looks like we should build the worker here too?

```suggestion
        "build:prod": "jlpm clean && jlpm build:lib:prod && jlpm build:worker && jlpm build:labextension",
```

**Comment by ianthomas23** [comment]
I think this is ready now. After merging this there is lots of work to be done on [cockle](https://github.com/jupyterlite/cockle).

**Pull request merged**